File: csharp/grade-school/GradeSchool.cs
using System.Collections.Generic;
using System.Linq;

public class GradeSchool
{
    private class Item
    {
      public string Student;
      public int Grade;
    };

    private readonly List<Item> _items = new List<Item>();

    public void Add(string student, int grade)
    {
      _items.Add(new Item
          {
            Student = student,
            Grade = grade
          }
      );
    }

    public IEnumerable<string> Roster()
    {
      return _items
        .OrderBy(i => i.Grade)
        .ThenBy(i => i.Student)
        .Select(i => i.Student);
    }

    public IEnumerable<string> Grade(int grade)
    {
        return _items
          .Where(i => i.Grade == grade)
          .OrderBy(i => i.Student)
          .Select(i => i.Student);
    }
}

File: python/grade-school/grade_school.py
import collections


class School:
    def __init__(self):
        self._students_by_grade = collections.defaultdict(list)

    def add_student(self, name: str, grade: int) -> None:
        self._students_by_grade[grade].append(name)

    def roster(self) -> list[str]:
        roster = []
        for grade in sorted(self._students_by_grade):
            roster += sorted(self._students_by_grade[grade])
        return roster

    def grade(self, grade: int) -> list[str]:
        return sorted(self._students_by_grade[grade])
